src/components/MaterialUI/DataDisplay/TransferList.js

```diff
@@ -0,0 +1,197 @@
+import React, { useState, useCallback } from "react";
+import { FormattedMessage } from "react-intl";
+import { makeStyles } from "@material-ui/core/styles";
+import Grid from "@material-ui/core/Grid";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemText from "@material-ui/core/ListItemText";
+import Button from "@material-ui/core/Button";
+import Paper from "@material-ui/core/Paper";
+import Typography from "@material-ui/core/Typography";
+import classNames from "classnames";
+import { isReactComponent } from "../../../utils/propertyValidator";
+import sharedMessages from "../../../sharedMessages";
+
+const useStyles = makeStyles(theme => ({
+	root: {
+		margin: "auto",
+	},
+
+	title: {
+		color: theme.palette.text.hint,
+		fontSize: theme.spacing(1.2),
+		marginBottom: theme.spacing(0.6),
+	},
+	paper: {
+		overflow: "auto",
+		width: theme.spacing(32.6),
+		height: theme.spacing(48),
+		border: `${theme.spacing(0.1)} solid ${theme.palette.primary.light}`,
+		borderRadius: theme.spacing(0.5),
+		boxShadow: "none",
+
+		"&::-webkit-scrollbar": {
+			width: theme.spacing(1.5),
+		},
+		"&::-webkit-scrollbar-thumb": {
+			background: theme.palette.grey.borders,
+			border: `${theme.spacing(0.5)} white solid`,
+			backgroundClip: "padding-box",
+			borderRadius: theme.spacing(1.5),
+		},
+	},
+	paperLeft: {
+		border: `${theme.spacing(0.1)} solid ${theme.palette.grey.borders}`,
+	},
+	item: {
+		fontSize: theme.spacing(1.3),
+		color: theme.palette.text.primary,
+		fontFamily: theme.typography.fontFamily,
+		padding: theme.spacing(0, 1.6),
+		cursor: "pointer",
+	},
+	selectedItem: {
+		backgroundColor: theme.palette.primary.light,
+	},
+
+	button: {
+		margin: theme.spacing(0.5, 0),
+		fontSize: theme.spacing(1.3),
+		lineHeight: theme.spacing(1.4),
+		padding: theme.spacing(0.7),
+		minWidth: theme.spacing(8),
+	},
+}));
+
+const formatState = (state, checked, from, to) => {
+	const fromSelectedItems = state[from].filter(({ id }) => checked.includes(id));
+	return {
+		[to]: [...state[to], ...fromSelectedItems].sort((a, b) => a.title.localeCompare(b.title)),
```

Comment:
I think we need to remove sort here - to able to have sort options from parent component which use this list. So we need to pass correct sorted items inside